symbolic-debuginfo/src/wasm.rs

```diff
@@ -0,0 +1,341 @@
+    pub fn has_unwind_info(&self) -> bool {
+        for (_, section) in self.wasm_module.customs.iter() {
+            if section.name() == ".debug_frame" {
+                return true;
+            }
+        }
+        false
+    }
+
+    /// Determines whether this object contains embedded source.
+    pub fn has_sources(&self) -> bool {
+        false
+    }
+
+    /// Returns the raw data of the WASM file.
+    pub fn data(&self) -> &'data [u8] {
+        self.data
+    }
+
+    /// Returns the offset of the code section.
+    pub fn code_offset(&self) -> u64 {
+        self.code_offset
+    }
+}
+
+impl fmt::Debug for WasmObject<'_> {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        f.debug_struct("WasmObject")
+            .field("code_id", &self.code_id())
+            .field("debug_id", &self.debug_id())
+            .field("arch", &self.arch())
+            .field("kind", &self.kind())
+            .field("load_address", &format_args!("{:#x}", self.load_address()))
+            .field("has_symbols", &self.has_symbols())
+            .field("has_debug_info", &self.has_debug_info())
+            .field("has_unwind_info", &self.has_unwind_info())
+            .finish()
+    }
+}
+
+impl<'slf, 'd: 'slf> AsSelf<'slf> for WasmObject<'d> {
+    type Ref = WasmObject<'slf>;
+
+    fn as_self(&'slf self) -> &Self::Ref {
+        self
+    }
+}
+
+impl<'d> Parse<'d> for WasmObject<'d> {
+    type Error = WasmError;
+
+    fn test(data: &[u8]) -> bool {
+        Self::test(data)
+    }
+
+    fn parse(data: &'d [u8]) -> Result<Self, WasmError> {
+        Self::parse(data)
+    }
+}
+
+impl<'data: 'object, 'object> ObjectLike<'data, 'object> for WasmObject<'data> {
+    type Error = DwarfError;
+    type Session = DwarfDebugSession<'data>;
+    type SymbolIterator = WasmSymbolIterator<'data, 'object>;
+
+    fn file_format(&self) -> FileFormat {
+        self.file_format()
+    }
+
+    fn code_id(&self) -> Option<CodeId> {
+        self.code_id()
+    }
+
+    fn debug_id(&self) -> DebugId {
+        self.debug_id()
+    }
+
+    fn arch(&self) -> Arch {
+        self.arch()
+    }
+
+    fn kind(&self) -> ObjectKind {
+        self.kind()
+    }
+
+    fn load_address(&self) -> u64 {
+        self.load_address()
+    }
+
+    fn has_symbols(&self) -> bool {
+        self.has_symbols()
+    }
+
+    fn symbols(&'object self) -> Self::SymbolIterator {
+        self.symbols()
+    }
+
+    fn symbol_map(&self) -> SymbolMap<'data> {
+        self.symbol_map()
+    }
+
+    fn has_debug_info(&self) -> bool {
+        self.has_debug_info()
+    }
+
+    fn debug_session(&self) -> Result<Self::Session, Self::Error> {
+        self.debug_session()
+    }
+
+    fn has_unwind_info(&self) -> bool {
+        self.has_unwind_info()
+    }
+
+    fn has_sources(&self) -> bool {
+        self.has_sources()
+    }
+}
+
+impl<'d> Dwarf<'d> for WasmObject<'d> {
+    fn endianity(&self) -> Endian {
```

Comment:
lol, only noticing this now, but isnt `endianness` the more common word for this?

Reply:
Blame @jan-auer